tsl/src/data_node.c

```diff
@@ -86,8 +86,22 @@ get_database_info(Oid dbid, DbInfo *database)
 	dbrecord = (Form_pg_database) GETSTRUCT(dbtuple);
 
 	database->encoding = dbrecord->encoding;
-	database->collation = dbrecord->datcollate;
-	database->chartype = dbrecord->datctype;
+
+#if PG15_GE
+	Datum datum;
+	bool isnull;
+
+	datum = SysCacheGetAttr(DATABASEOID, dbtuple, Anum_pg_database_datcollate, &isnull);
```

Comment:
I would suggest to mention the changed lookup of the attributes in the commit message.